### PR TITLE
Middle.mouse.click features

### PR DESCRIPTION
**Pull request by Bortseb**
First of all, woops, didn't realize I had other code tagging along for the ride on this one...

Maybe I can just merge this and then fix the glitchy panning next?

Anyhow, here is what I've added to middle mouse click:

- If you middle mouse click on a topic, or selection of topics, it will open all of the links within those topics in a new browser tab(s) (i.e. you don't need to go into the topic card.... this is for my "bookmarking" use-case)

- Also, I used the code I recently saw Devin add for sending text to the clipboard... and made it so that if you hold CTRL and middle mouse click on a topic or a synapse, then its label gets added to your clipboard.

I have more things that I want to add, but I couldn't get around to them yet, see below:
image

### CODE REVIEW

**Comment by devvmh** [comment]
This looks cool! I'm loving the features. Middle click + Ctrl is very power usery, but I'll use it I'll bet.

The middle click to open all links in tabs is just so great. Adding it together with import... man, Metamaps is gonna be a cool tool come Oct 31

As far as merging goes, I think you need to rebase this on develop anyways. I can help with that from my end if it's too complicated on c9. It would be good for the history to have a clean PR/commit that shows exactly what was included in this feature

**Comment by Bortseb** [comment]
Just cleaning up what issues are assigned to me, and seeing how I might help next... Did you get a chance to look over this one @Connoropolous ?

File: frontend/src/Metamaps/Map/index.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ const Map = {
     $('#wrapper').on('contextmenu', function (e) {
       return false
     })
+    
+    $('#wrapper').mousedown(function (e){
+      if(e.button==1)return false;
```
Comment (Connoropolous) [inline frontend/src/Metamaps/Map/index.js]: what's this?

File: frontend/src/Metamaps/JIT.js [hunk 2]
```diff
@@ -1489,6 +1509,9 @@ const JIT = {
     if (Visualize.mGraph.busy) return
 
     const self = JIT
+    var synapseText = adj.data.$synapses[0].attributes.desc;
+    //Copy synapse label to clipboard
+    if(e.button===1 && e.ctrlKey && synapseText != "") clipboard.copy(synapseText);
```
Comment (Connoropolous) [inline frontend/src/Metamaps/JIT.js]: use !==

File: frontend/src/Metamaps/JIT.js [hunk 3]
```diff
@@ -1245,25 +1245,45 @@ const JIT = {
...
-          if (!e.shiftKey) {
-            Control.deselectAllNodes()
-            Control.deselectAllEdges()
-          }
-
-          if (nodeAlreadySelected) {
-            Control.deselectNode(node)
+          var nodeAlreadySelected = node.selected
+          
+          if( !(e.button===1) ){
```
Comment (Connoropolous) [inline frontend/src/Metamaps/JIT.js]: use e.button !== 1

File: frontend/src/Metamaps/Map/index.js (hunk 1 above)
Comment (Bortseb) [inline frontend/src/Metamaps/Map/index.js]: You know when you click in the middle mouse, there is a default action in some browsers to star letting you 'pan' around the page... plus the pointer icon changes... this prevents that

Comment (Connoropolous) [inline frontend/src/Metamaps/Map/index.js]: cool, I see. ===

**Comment by Connoropolous** [comment]
ping apparently reviews don't show in slack

**Comment by Connoropolous** [comment]
actually, there's a merge conflict apparently, connected with the other stuff you merged in I think. that'll have to get resolved first. prbly very small

**Comment by Bortseb** [comment]
its likely simple... but I made all the changes just on github... now im confused how to find the conflict using command line

can I pull in whats currently here?

**Comment by Connoropolous** [comment]
there it is. sweet! merge it

**Pull request merged**